Src/FluentAssertions/Common/TypeExtensions.cs

```diff
@@ -31,13 +31,10 @@ public static bool IsDecoratedWith<TAttribute>(this Type type)
         return type.IsDefined(typeof(TAttribute), inherit: false);
     }
 
-    public static bool IsDecoratedWith<TAttribute>(this MemberInfo type)
+    public static bool IsDecoratedWith<TAttribute>(this ICustomAttributeProvider type)
         where TAttribute : Attribute
     {
-        // Do not use MemberInfo.IsDefined
-        // There is an issue with PropertyInfo and EventInfo preventing the inherit option to work.
-        // https://github.com/dotnet/runtime/issues/30219
-        return Attribute.IsDefined(type, typeof(TAttribute), inherit: false);
+        return type.IsDefined(typeof(TAttribute), inherit: false);
```

Comment:
It seems this will be calling the `MemberInfo.IsDefined` that the comment said _not_ to use.
I dug into this and the related PRs are #748 and #1095.

I checked out develop at #748 and used the prohibited method.
I don't have net45 or netcoreapp1.2 installed, but for the remaining TFMs no tests failed.
Since we no longer support these TFMs directly or indirectly, I think it's safe to use the `ICustomAttributeProvider.IsDefined`.